api/provider/aws/common.go

```diff
@@ -12,29 +12,68 @@ import (
 	"github.com/aws/aws-sdk-go/service/ecs/ecsiface"
 	"github.com/aws/aws-sdk-go/service/elb"
 	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+	alb "github.com/aws/aws-sdk-go/service/elbv2"
+	albiface "github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
 	"github.com/quintilesims/layer0/api/tag"
 	"github.com/quintilesims/layer0/common/errors"
 )
 
-func describeLoadBalancer(elbapi elbiface.ELBAPI, loadBalancerName string) (*elb.LoadBalancerDescription, error) {
-	input := &elb.DescribeLoadBalancersInput{}
-	input.SetLoadBalancerNames([]*string{aws.String(loadBalancerName)})
-	input.SetPageSize(1)
+// searches for the load balancer name as both classic and application load balancers and returns
+// the first found result or an error if the neither classic or application lb could be found for
+// the given lb name
+func describeLoadBalancer(elbapi elbiface.ELBAPI, albapi albiface.ELBV2API, loadBalancerName string) (*genericLoadBalancer, error) {
+	// search classic elastic load balancers
+	searchELB := func() (*genericLoadBalancer, error) {
```

Comment:
What do we gain by bundling the ELB search logic into an interior function? I don't think it's used more than once, and I'm not seeing any logic that would be particularly simplified. Maybe the `if err and err != LoadBalancerNotFound` bit, but ALB search just below has that logic as well, and isn't bundled into a function. It just seems strange to me that ELB search and ALB search are so incredibly similar, but are treated pretty differently.